mlrun/serving/states.py

```diff
@@ -775,7 +775,13 @@ def add_route(
             2. Create a new model endpoint with the same name and set it to `latest`.
 
         """
-
+        if len(self.routes.keys()) >= MAX_MODELS_PER_ROUTER and key not in self.routes:
+            raise mlrun.errors.MLRunModelLimitExceededError(
+                f"Router cannot support more than {MAX_MODELS_PER_ROUTER} model endpoints. "
+                f"To add a new route, edit an existing one by passing the same key."
+            )
+        if key in self.routes:
```

Comment:
Minor optimization: if you do this check first, you can do the second check in the "else" clause withuot the "not in self.routes"